clock.go

```diff
@@ -20,13 +20,14 @@
 
 package zap
 
-import "time"
+import (
+	"time"
+
+	"go.uber.org/zap/zapcore"
+)
 
 // Clock is a source of time for logged entries.
-type Clock interface {
-	// Now returns the current local time.
-	Now() time.Time
-}
+type Clock zapcore.Clock
```

Comment:
Do we need to keep the zap-level Clock type? We haven't shipped this yet so we can move this type to zapcore without breaking anything.

Also, if we do want to keep it around, we probably want to use a type alias.

```suggestion
type Clock = zapcore.Clock
```

The current form declares a new type Clock that is distinct from zapcore.Clock, while the type alias resolves to zapcore.Clock at compile time.